### PR TITLE
FBE-1442: Update MAPI /v1/event/purchase documentation for addition of referralCode parameter.

### PR DESCRIPTION
**Pull request by dhull**
* Update README.md to describe how Friendbuy uses Slate.

* Update MAPI /v1/event/purchase documentation for addition of referralCode parameter.

### CODE REVIEW

File: source/includes/_merchant_api.md [hunk 1]
```diff
@@ -719,11 +720,11 @@ When a Friendbuy referral link is followed, Friendbuy adds the GET parameter `fb
 
 The Friendbuy Merchant API can be used to send purchase events to Friendbuy as an alternative or supplement to our Javascript-based pixel tracking. Common use cases include tracking purchases that occur offline and purchases that occur within a mobile app.
 
-When you submit a purchase using the Friendbuy Merchant API, you must submit a unique order id, a purchase amount, and the currency the purchase was in. We strongly recommend passing the email and/or customer id of the user who made the purchase. We also recommend passing a value for isNewCustomer, which we use as part of our reward criteria when determining if the advocate should earn a reward.
+When you submit a purchase using the Friendbuy Merchant API, you must submit a unique order id, a customer id, a purchase amount, and the currency the purchase was in. We strongly recommend also passing the email address of the customer who made the purchase. We also recommend passing a value for isNewCustomer, which we use as part of our reward criteria when determining if the advocate should earn a reward.
```
Comment (britainjs) [inline source/includes/_merchant_api.md:723]: I believe customer id is an optional property since a lot of merchants support guest checkout.

**Review by britainjs — CHANGES_REQUESTED** [review]
One note. Looks good otherwise.

File: source/includes/_merchant_api.md (hunk 1 above)
Comment (dhull) [reply source/includes/_merchant_api.md:723]: I was updating this according to what the openapi.yaml says.

```yaml
    purchaseEventRequest:
      type: object
      required:
        - orderId
        - customerId
        - amount
        - currency
```

Comment (britainjs) [reply source/includes/_merchant_api.md:723]: I did some more poking around and it doesn't actually seem like there is a way to track purchases with an email but no customer id, either through the mapi or through javascript. https://kibana.fbot.me/_plugin/kibana/goto/7b4f9583fbb80d7f76e937054ae45bf4 

Definitely out of the scope for this project, but something we should address.

Comment (dhull) [reply source/includes/_merchant_api.md]: I have reverted the change to say that customerId is required. We still need to update the openapi.yaml for that change.

**Review by britainjs — APPROVED** [review]

**Pull request merged**